qemu/tests/block_transfer_parameters_check.py

```diff
@@ -0,0 +1,155 @@
+"""Verify Maximum transfer length and max_sector_kb in guest """
+
+import logging
+import time
+from os.path import basename
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import data_dir
+from virttest import utils_misc
+
+from virttest.iscsi import Iscsi
+from virttest.utils_misc import get_linux_drive_path
+import json
+import copy
+
+
+def run(test, params, env):
+    """
+    Verify Maximum transfer length and max_sector_kb in guest.
+
+    Steps:
+        1) Get max_sector_kb and segments of host target device.
+        2) Pass-through host target device and Boot VM.
+        3) Login guest and check the max_sector_kb and Maximum transfer length.
+        4) Execute IO Check the VM still running.(optional)
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def _verify_transfer_info(host_info, guest_info):
+        errmsg = "The guest get unexpected transfer %s" % guest_info
+        if guest_info["sectors_kb"] > host_info["sectors_kb"]:
+            test.fail(errmsg)
+        min_host = min(host_info["sectors_kb"], host_info["segments"] * 4)
+        logger.debug(min_host)
+        if guest_info["sectors_kb"] > min_host:
+            test.fail(errmsg)
+        if guest_info["tran_length"] > guest_info["sectors_kb"] * 2:
+            test.fail(errmsg)
+
+    def _get_target_devices(trans_type, size=None):
+        devs = []
+        cond = "$2==\"%s\"" % trans_type
+        if size:
+            cond = cond + " && $3==\"%s\"" % size
+        cmd = "lsblk -Spo 'NAME,TRAN,SIZE' |awk '{if(%s) print $1}'" % cond
+        status, output = process.getstatusoutput(cmd)
+        devs_str = output.strip().replace("\n", " ")
+        if devs_str:
+            cmd = "lsblk -Jpo 'NAME,HCTL,SERIAL,TRAN,FSTYPE,WWN' %s" % devs_str
+            status, output = process.getstatusoutput(cmd)
+            devs = copy.deepcopy(json.loads(output)["blockdevices"])
+
+        return devs
+
+    def _run_cmd(cmd, conn=None):
+        if conn:
+            return conn.cmd_output(cmd)
+        else:
+            return process.getoutput(cmd)
+
+    def _get_transfer_parameters(dev, conn=None):
+        dev = basename(dev)
+        info = {"tran_length": 0, "sectors_kb": 0, "segments": 0}
+        cmd = params["get_tran_length_cmd"] % dev
+        output = _run_cmd(cmd, conn)
+        logger.debug(output)
+        info["tran_length"] = int(output)
+        cmd = params["get_tran_params_cmd"] % dev
+        output = _run_cmd(cmd, conn).split()
+        logger.debug(output)
+        info["sectors_kb"] = int(output[0])
+        info["segments"] = int(output[1])
+        return info
+
+    vm = None
+    iscsi = None
+    logger = logging.getLogger('avocado.test')
```

Comment:
FYI. The test object has it,  e.g. test.log.debug, test.log.info. So we could use them directly.  Thanks.

Reply:
update to logger= test.log